chain/sync.go

```diff
@@ -1279,3 +1340,24 @@ func (syncer *Syncer) MarkBad(blk cid.Cid) {
 func (syncer *Syncer) CheckBadBlockCache(blk cid.Cid) (string, bool) {
 	return syncer.bad.Has(blk)
 }
+func (syncer *Syncer) getLatestBeaconEntry(ctx context.Context, ts *types.TipSet) (*types.BeaconEntry, error) {
+	cur := ts
+	for i := 0; i < 20; i++ {
```

Comment:
That number is speced, right? If yes, then pull into build / grab from specs-actors if it will be there. (if not then probably bug spec about it)

Reply:
yeah, this is probably a spec thing. theres no number defined though.

Reply:
technically, the maximum number of blocks that can not have drand values in them is defined by the drand interval, we could pass in the beacon object and compute that using it